src/main/java/com/redhat/red/build/finder/BuildFinder.java

```diff
@@ -461,6 +463,38 @@ public static void main(String[] args) {
                 throw new ParseException("Must specify at least one file");
             }
 
+            if (line.hasOption("debug")) {
+                ch.qos.logback.classic.Logger rootLogger = (ch.qos.logback.classic.Logger) LoggerFactory.getLogger(org.slf4j.Logger.ROOT_LOGGER_NAME);
+                rootLogger.setLevel(Level.DEBUG);
+
+                LoggerContext loggerContext = rootLogger.getLoggerContext();
+                loggerContext.reset();
+
+                ((ch.qos.logback.classic.Logger) LoggerFactory.getLogger("com.redhat.red.build.koji")).setLevel(Level.WARN);
+                ((ch.qos.logback.classic.Logger) LoggerFactory.getLogger("org.apache.commons.beanutils")).setLevel(Level.WARN);
+                ((ch.qos.logback.classic.Logger) LoggerFactory.getLogger("org.apache.commons.vfs2")).setLevel(Level.WARN);
+                // XXX: Set to OFF due to <https://issues.apache.org/jira/browse/VFS-634>
+                ((ch.qos.logback.classic.Logger) LoggerFactory.getLogger("org.apache.commons.vfs2.impl")).setLevel(Level.OFF);
+                ((ch.qos.logback.classic.Logger) LoggerFactory.getLogger("org.apache.http")).setLevel(Level.WARN);
+                ((ch.qos.logback.classic.Logger) LoggerFactory.getLogger("org.apache.kerby")).setLevel(Level.WARN);
+                ((ch.qos.logback.classic.Logger) LoggerFactory.getLogger("org.commonjava.util.jhttpc")).setLevel(Level.WARN);
+                ((ch.qos.logback.classic.Logger) LoggerFactory.getLogger("org.commonjava.rwx")).setLevel(Level.WARN);
+
+                PatternLayoutEncoder encoder = new PatternLayoutEncoder();
+                encoder.setContext(loggerContext);
+                encoder.setPattern("%d{HH:mm:ss.SSS} [%thread] %-5level %logger{36} - %msg%n");
+                encoder.start();
+
+                ConsoleAppender<ILoggingEvent> appender = new ConsoleAppender<ILoggingEvent>();
+                appender.setContext(loggerContext);
+                appender.setEncoder(encoder);
+                appender.start();
+
+                rootLogger.addAppender(appender);
+            }
```

Comment:
While it seems quite straightforward to configure different levels within the logback.xml it doesn't seem so obvious if we have a different root level configuration to have a different patternlayout. For simplicity this should be fine. If we remove the reset you'll probably need to add `rootLogger.detachAppender("STDOUT");` to avoid duplicate lines (and maybe STDOUT should be renamed to e.g. default?)

Reply:
It may be possible to inherit the current configuration from logback.xml and to change the just pattern rather than resetting everything and repeat it again.

The reset was to avoid getting both configurations mixed and thereby getting duplicate lines. Because of the reset, repeating the entire configuration was necessary.